modules/instrument_list/php/instrument_list.class.inc

```diff
@@ -134,6 +134,29 @@ class Instrument_List extends \NDB_Menu_Filter
             // the candidate exists before getting here.
             throw new \LorisException("Timepoint list requires a timepoint.");
         }
+
+        // Check imaging_browser permission, used to control imaging_browser link
+        // on the instrument_list main page
+        $loris = $request->getAttribute("loris");
+        $perm  = false;
+        if (!is_null($loris) && $loris->hasModule('imaging_browser')) {
```

Comment:
Hey @lingma, this is the first time I'm seeing the `hasModule` used and from the `$request->getAttribute("loris");` object and I'm wondering if it would be better to use the following code:
```
$user =& User::singleton();
$user->hasPermission('imaging_browser');
```
I'm also unsure if it's secure to do permission checks on the backend from frontend objects? Maybe @johnsaigle or @driusan can comment. My apologies if I'm mistaken. Otherwise the code looks good.

Reply:
Hi @maltheism the `hasModule` from the `$request->getAttribute("loris");` object is to check whether the module in question is active or not and the module file exists or not, which is not the same as check if the user has permission or not. This kind of check is new and only exists in the master branch.

Reply:
thanks @lingma for the explanation. I'll manually test the branch and give the approval soon.

Reply:
@maltheism This is almost the same code as the PR #6365 which has been approved by @nicolasbrossard, the method in the PR #6365 is faster than the method here but the previous added some complexity.

Reply:
I believe there should also be a check in there that the user has the appropriate project for the button to be present.

Reply:
@driusan @maltheism I think we should also check whether the user belongs to the appropriate project before displaying the button. In fact, the _entire page_ should not be displayed if the user does not have the appropriate project (which is not the case right now).

Reply:
The project verification should be a standard function from MenuFilter. If the project verification doesn't work, then the filter configuration is not properly done. I believe the project verification is not part of the purpose of this PR.

Reply:
I am talking about the instrument list page, which is not a `MenuFilter` anyway. Even if it was, I think that the project-based filtering is not performed in order to find out whether a particular page can be displayed/accessed or not: the `hasAccess()` method should do that I think.

You are right when you say that this might be outside this PR's mandate. Just wanted to raise the issue.

Reply:
`instrument_list` inherits `NDB_Menu_Filter`, project level filter is done by inheritance as I remember, I suggest you to report this bug as an issue.

Reply:
I have difficulty to test as my old installation doesn't work well, I have not tested the issue you reported, I think you can report as an issue.

Reply:
I wrongly assumed `Menu_Filter` as `DataFrameworkMenu`, the project is not filtered, you can modify instrument_list to inherit `DataFrameworkMenu`, then add a `useProjectFilter`, which might disable `instrument_list`'s project filter function in the user interface. As a result, I believe that not to filter is on purpose by system design, there is not any need to add project control unless you want to modify a lot more. It is up to you and @driusan

Reply:
reported in #6928

Reply:
Hi guys please see PR #6836 and (issue: #6829) -- it's a known issue with a solution committed a month ago.
it's awaiting input from Dave/core team.

Reply:
They are not the same, #6836 (#6829) talks about user permission, @nicolasbrossard talked about project filter, though I believe it is not necessary to add project filter on individual instrument.